src/test/java/org/jabref/logic/util/UpdateFieldTest.java

```diff
@@ -167,4 +172,114 @@ public void testUpdateNonDisplayableFieldHasNotChanged() {
+
+        assertNotEquals(entry.getField(StandardField.OWNER).get(), "testDefaultOwner", "Owner has changed");
+        assertEquals(entry.getField(StandardField.OWNER).get(), alreadySetOwner, "Owner has not changed");
+    }
+
+    @Test
+    public void emptyCreationdateFieldNowPresentAfterAutomaticSet() {
+        assertFalse(entry.getField(StandardField.CREATIONDATE).isPresent(), "CreationDate is present");
+
+        OwnerPreferences ownerPreferences = createOwnerPreference(true, true);
+        TimestampPreferences timestampPreferences = createTimestampPreference();
+        UpdateField.setAutomaticFields(entry, false, false, ownerPreferences, timestampPreferences);
+
+        assertTrue(entry.getField(StandardField.CREATIONDATE).isPresent(), "No CreationDate exists");
+    }
+
+    @Test
+    public void creationdateAssignedCorrectlyAfterAutomaticSet() {
+        OwnerPreferences ownerPreferences = createOwnerPreference(true, true);
+        TimestampPreferences timestampPreferences = createTimestampPreference();
+        UpdateField.setAutomaticFields(entry, false, false, ownerPreferences, timestampPreferences);
+
+        String creationDate = timestampPreferences.now();
+
+        assertEquals(entry.getField(StandardField.CREATIONDATE).get(), creationDate, "Not the same date");
+    }
+
+    @Test
+    public void ownerSetToDefaultValueForCollectionOfBibEntries() {
+        BibEntry entry2 = new BibEntry();
+        BibEntry entry3 = new BibEntry();
+
+        assertFalse(entry.getField(StandardField.OWNER).isPresent(), "Owner field for entry is present");
+        assertFalse(entry2.getField(StandardField.OWNER).isPresent(), "Owner field for entry2 is present");
+        assertFalse(entry3.getField(StandardField.OWNER).isPresent(), "Owner field for entry3 is present");
+
+        Collection<BibEntry> bibs = Arrays.asList(entry, entry2, entry3);
+
+        OwnerPreferences ownerPreferences = createOwnerPreference(true, true);
+        TimestampPreferences timestampPreferences = createTimestampPreference();
+        UpdateField.setAutomaticFields(bibs, false, ownerPreferences, timestampPreferences);
+
+        String defaultOwner = "testDefaultOwner";
+
+        assertEquals(entry.getField(StandardField.OWNER).get(), defaultOwner, "entry has no owner field");
+        assertEquals(entry2.getField(StandardField.OWNER).get(), defaultOwner, "entry2 has no owner field");
+        assertEquals(entry3.getField(StandardField.OWNER).get(), defaultOwner, "entry3 has no owner field");
+    }
+
+    @Test
+    public void ownerNotChangedForCollectionOfBibEntriesIfOptionsDisabled() {
+        String initialOwner = "initialOwner";
+
+        entry.setField(StandardField.OWNER, initialOwner);
+        BibEntry entry2 = new BibEntry();
+        entry2.setField(StandardField.OWNER, initialOwner);
+
+        assertTrue(entry.getField(StandardField.OWNER).isPresent(), "Owner field for entry is not present");
```

Comment:
```suggestion
        assertEquals(Optional.of(initialOwner), entry.getField(StandardField.OWNER));
```

Similar for all other Optional tests. We should avoid `assertTrue` in all cases; we should use `assertEquals` wherevery possible.